geolocator_android/android/src/main/java/com/baseflow/geolocator/MethodCallHandlerImpl.java

```
@@ -238,7 +258,25 @@ private void onGetCurrentPosition(MethodCall call, MethodChannel.Result result)
 
           replySubmitted[0] = true;
           geolocationManager.stopPositionUpdates(locationClient);
+          pendingCurrentPositionLocationClients.remove(requestId);
           result.error(errorCode.toString(), errorCode.toDescription(), null);
         });
   }
+
+  /**
+   * Cancels a request for the current position that was initiated with {@link #onGetCurrentPosition}.
+   */
+  private void onCancelGetCurrentPosition(MethodCall call, MethodChannel.Result result) {
+    @SuppressWarnings("unchecked")
+    Map<String, Object> arguments = (Map<String, Object>) call.arguments;
+    String requestId = (String) arguments.get("requestId");
+
+    LocationClient locationClient = this.pendingCurrentPositionLocationClients.get(requestId);
+    if (locationClient != null) {
+      locationClient.stopPositionUpdates();
+    }
+    this.pendingCurrentPositionLocationClients.remove(requestId);
+
+    result.success(null);
```

Fail silently in case there is no location client for the request identifier, or call result.error()?

---

In this case I think failing silently would be the right approach. It should never happen and if for some reason it does happen returning an error will only cause confusion to the consuming developer.